Adding sessions uptime

kubatom opened the pull request:
Adding sessions uptime in the same format as the usage. Updated README.md and left no breaking changes to the code. 

I would suggest changing routing to add a specific endpoint for get_usage/target= instead of  /target=

kubatom committed:
Adding sessions uptime exporter

kubatom committed:
Merge pull request #1 from kubatom/sessions_uptime

Adding sessions uptime exporter

TheHolm commented on per-user-usage.py:
```diff
@@ -1,118 +1,230 @@
                 thread_data.uid = str(thread_data.varBind[0]).split(".")
                 try:
-                    interface_IDs[str(thread_data.varBind[1])] = thread_data.uid[-2]+'.'+thread_data.uid[-1]
+                    interface_IDs[str(thread_data.varBind[1])] = str(thread_data.uid[-2] + '.' + thread_data.uid[-1])
```
Why cast to str after + operator on strings?  It will be string type anyway, or file on int to str conversion.

kubatom replied:
It should be, there was some weird behaviour for it that you wrapped in try and except statement. I wanted to investigate it further, but you can ignore this change.

TheHolm commented on per-user-usage.py:
```diff
@@ -153,40 +268,51 @@ def do_GET(self):
-                responce = responce + 'ifInOctets{ user="' + username + '" } ' + str(users_stats[username]['RX_Octets']) + '\n'
-            except KeyError:
-                pass
+        if "/get_sessions_uptime/?target=" in self.path:
```
NO new URL is required, uptime data should be returned at same time as RX/TX data.

TheHolm commented on per-user-usage.py:
```diff
@@ -1,118 +1,230 @@
+    print(f"Got {len(sessions_uptime)} session uptime records for OID: {OID}")
+
+
+def get_sessions_uptime(auth_data, SNMP_target):
```
please incorporate it to "get_usage(auth_data, SNMP_target)", just add one more thread and use already collected "get_l2tp_users" and "interface_IDs"

TheHolm commented on per-user-usage.py:
```diff
@@ -1,118 +1,230 @@
+                thread_data.uid = str(thread_data.varBind[0]).split(".")
+                try:
+                    time_delta_string = convert_timetick_to_datetime(int(thread_data.varBind[1]))
+                    sessions_uptime[str(thread_data.uid[-2] + '.' + thread_data.uid[-1])] = str(time_delta_string)
```
prometheus  can't collect texts.  only floats.

kubatom replied:
resolved in the latest commit

kubatom committed:
Adding changes as requested:

- removing formatting to str
- removed path as requested
- incorporated all the code to one function, added extra thread
